new-installer/targets/aws/iac/kms/kms_test.go

```diff
@@ -0,0 +1,148 @@
+// SPDX-FileCopyrightText: 2025 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package aws_iac_kms_test
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	aws_sdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/gruntwork-io/terratest/modules/aws"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/suite"
+)
+
+func policyString(t *testing.T, clusterName string) string {
+	t.Helper()
+	parsedAccId := strings.ReplaceAll(policy, "${local.account_id}", aws.GetAccountId(t))
+	return strings.ReplaceAll(parsedAccId, "${aws_iam_user.vault.name}", "vault-"+clusterName)
+}
+
+const policy string = `{
+    "Id": "vault",
+    "Statement": [
+        {
+            "Sid": "Enable IAM User Permissions",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::${local.account_id}:root"
+            },
+            "Action": "kms:*",
+            "Resource": "*"
+        },
+        {
+            "Sid": "Allow use of the key",
+            "Action": [
+                "kms:Encrypt",
+                "kms:Decrypt",
+                "kms:ReEncrypt*",
+                "kms:GenerateDataKey*",
+                "kms:DescribeKey"
+            ],
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::${local.account_id}:user/${aws_iam_user.vault.name}"
+            },
+            "Resource": "*"
+        }
+    ],
+    "Version": "2012-10-17"
+}`
+
+type KMSTestSuite struct {
+	suite.Suite
+}
+
+type KMSVariables struct {
+	Region      string `json:"region" yaml:"region"`
+	CustomerTag string `json:"customer_tag" yaml:"customer_tag"`
+	ClusterName string `json:"cluster_name" yaml:"cluster_name"`
+}
+
+func TestKMSTestSuite(t *testing.T) {
+	suite.Run(t, new(KMSTestSuite))
+}
+
+func (s *KMSTestSuite) TestApplyingModule() {
+	randomPostfix := strings.ToLower(rand.Text()[:4])
+	bucketName := "kms-test-bucket-" + randomPostfix
+	aws.CreateS3Bucket(s.T(), "us-west-2", bucketName)
+	defer func() {
+		aws.EmptyS3Bucket(s.T(), "us-west-2", bucketName)
+		aws.DeleteS3Bucket(s.T(), "us-west-2", bucketName)
+	}()
+	clusterName := "kms-test-" + randomPostfix
+	variables := KMSVariables{
+		Region:      "us-west-2",
+		CustomerTag: "test-customer",
```

Comment:
Use the one from utils module

```suggestion
		Region: utils.DefaultTestRegion,
		CustomerTag: utils.DefaultTestCustomerTag,
```